src/metatensor/models/utils/data/combine_dataloaders.py

```diff
@@ -1,57 +1,58 @@
-import itertools
 from typing import List
 
 import numpy as np
 import torch
 
 
-class CombinedIterableDataset(torch.utils.data.IterableDataset):
+class CombinedDataLoader:
     """
-    Combines multiple dataloaders into a single iterable dataset.
-    This is useful for combining multiple dataloaders into a single
-    dataloader. The new dataloader can be shuffled or not.
-
-    :param dataloaders: list of dataloaders to combine
-    :param shuffle: whether to shuffle the combined dataloader
+    Combines multiple dataloaders into a single dataloader.
 
-    :return: combined dataloader
+    This is useful for learning from multiple datasets at the same time,
+    each of which may have different batch sizes, properties, etc.
     """
 
-    def __init__(self, dataloaders, shuffle):
+    def __init__(self, dataloaders: List[torch.utils.data.DataLoader], shuffle: bool):
+        """Creates the combined dataloader.
+
+        :param dataloaders: list of dataloaders to combine
+        :param shuffle: whether to shuffle the combined dataloader (this does not
+            act on the individual batches, but it shuffles the order in which
+            they are returned)
+
+        :return: the combined dataloader
+        """
```

Comment:
There should be no docstring for the init. Everything should be in the class docstring.